### DISCUSSION TITLE
Use file-list which is encoded with UTF-8

### DISCUSSION THREAD

**Discussion by flarens** [Yutaka-Sawada/MultiPar#141] [Q&A] [answered]
Thanks for the great work!

What is the correct way to use an incoming list file? I created a list.txt list with this content:

```
C:\test1.zip
C:\test2.zip
```

I call the command line to create recovery files:

`par2j64.exe c -rf1 -rr10 C:\bd.par2 -fu C:\list.txt`

And I get an error:

`input file is not found, -fu`

How should it work to pass a list of all files to be processed as a single list file? There are no examples anywhere. Why didn't you make a minimal page with examples of how to use each command, so as not to torture users with guesswork and so as not to distract you with such basic questions?

**Comment by Yutaka-Sawada** [comment, marked as answer]
> How should it work to pass a list of all files to be processed as a single list file?

To distinguish filenames from options, you must put all options before filenames. Because you put the option after `par filename`, the `-fu` was thought as an filename. When you set options in correct order, it's like `par2j64.exe c -rf1 -rr10 -fu C:\bd.par2 C:\list.txt`.

> Why didn't you make a minimal page with examples of how to use each command, so as not to torture users with guesswork and so as not to distract you with such basic questions?

It's because par2j is open source. When someone wants to know how it works, he just reads the source code. Even when I write a detailed manual for par2j, most users won't read it. If you want to see an example of command-line, you may read MultiPar's log file.

**Reply by flarens** [reply]
Thanks. Your suggested command line still doesn't see the list file for some reason:

`cannot open file-list, С:\list.txt`

Great tip to run the logs in the MultiPar program, I can see all commands

**Reply by Yutaka-Sawada** [reply]
> Your suggested command line still doesn't see the list file for some reason:

The error is caused by Windows OS. Though I don't know your using Windows OS version and settings, it may be forbidden to use a file direct under C drive by security risk. If you are using Windows 10 or Windows 11, you will need to change system settings. I suggest to put files in a folder, like `C:\test\list.txt`.

**Reply by flarens** [reply]
It's amazing. I have been a programmer all my life and this is the first time I have encountered the fact that Windows does not give permission to read files in the root of the C drive. Now it works as required.
